h5py/_hl/dataset.py

```diff
@@ -396,6 +396,14 @@ def size(self):
             self._cache_props['size'] = size
         return size
 
+    @property
+    def nbytes(self):
+        """Numpy-style attribute giving the dataset size as the number of bytes"""
+        size = self.size
+        if size is None:  # if we are empty, return None
+            return None
+        return numpy.dtype(self.dtype).itemsize * size
```

Comment:
I think `self.dtype` should always be a numpy dtype object already.

Reply:
```suggestion
        return self.dtype.itemsize * size
```